script/setup_heroku

```diff
@@ -1,45 +1,49 @@
-#!/bin/sh
-
-set -e
+#!/bin/bash
+
+set -ue
+
+function create_heroku_addon () {
+	if heroku addons:info "$1" &> /dev/null; then
+		echo "Already installed $1"
+	else
+		echo "Installing $1"
+		heroku addons:create "$@"
+	fi
+}
+
+if [ $# -eq 0 ]; then
+	echo >&2 "Please add your Heroku app name to the end of the command."
+	echo >&2 "Usage: setup_heroku your_app_name"
+	exit 1
+fi
 
-if [ $1 ]
-then
-  herokuApp="$1"
+export HEROKU_APP="$1"
 
-  echo "Getting ready to set environment variables for $herokuApp"
-  figaro heroku:set -e production --app $herokuApp
+echo "Getting ready to set environment variables for $HEROKU_APP"
+figaro heroku:set -e production
 
-  echo "Setting MAILER_URL"
-  heroku config:set MAILER_URL=$herokuApp.herokuapp.com --app $herokuApp
+echo "Setting MAILER_URL"
+heroku config:set MAILER_URL="$HEROKU_APP.herokuapp.com"
 
-  echo "Setting DEVISE_SECRET_KEY"
-  # generate a random string with 36 characters
-  token1=$(python -c 'import uuid; print (uuid.uuid4())')
-  heroku config:set DEVISE_SECRET_KEY=$token1 --app $herokuApp
+if [ -z "$(heroku config:get SECRET_TOKEN)" ]; then
+	echo "Setting SECRET_TOKEN"
+	# generate a random string with 36 characters
+	heroku config:set SECRET_TOKEN="$(ruby -rsecurerandom -e 'puts SecureRandom.uuid')"
```

Comment:
I think you misunderstood. I agree with you. I was saying that before, we wanted to support use cases where you would run the script without installing the app's dependencies. What I'm saying now is that I wouldn't be opposed to dropping support for those scenarios. I think it's reasonable to ask people to run `bin/setup` first before being able to set up Heroku. That way, `rake secret` will be available.